openquake/hmtk/seismicity/catalogue.py

```diff
@@ -107,8 +115,38 @@ def get_number_events(self):
     def add_event(self):
         raise NotImplementedError
 
-    def write_catalogue(self, output_file, filetype):
-        raise NotImplementedError
+    def write_catalogue(self, output_file, key_list=SORTED_ATTRIBUTE_LIST):
+        """
+        Writes the catalogue to file using HTMK format (CSV).
+
+        :param output_file:
+            Name of the output file
+        :param key_list:
+            Optional list of attribute keys to be exported
+        """
+
```

Comment:
It is better to write this as a with statement, `with open(output_file, 'wt') as outfile`.